ouroboros-network-api/src/Ouroboros/Network/NodeToNode/Version.hs

```diff
@@ -176,6 +181,46 @@ nodeToNodeCodecCBORTerm version
         decodeTerm _ t
           = Left $ T.pack $ "unknown encoding: " ++ show t
      in CodecCBORTerm {encodeTerm, decodeTerm = decodeTerm version }
+  | version >= NodeToNodeV_11 =
+    let encodeTerm :: NodeToNodeVersionData -> CBOR.Term
+        encodeTerm NodeToNodeVersionData { networkMagic, diffusionMode, peerSharing }
+          = CBOR.TList $
+              [ CBOR.TInt (fromIntegral $ unNetworkMagic networkMagic)
+              , CBOR.TBool (case diffusionMode of
+                             InitiatorOnlyDiffusionMode         -> True
+                             InitiatorAndResponderDiffusionMode -> False)
+              , CBOR.TInt (case peerSharing of
+                             NoPeerSharing      -> 0
+                             PeerSharingPrivate -> 1
+                             PeerSharingPublic  -> 2)
+              ]
+
+        decodeTerm :: NodeToNodeVersion -> CBOR.Term -> Either Text NodeToNodeVersionData
+        decodeTerm _ (CBOR.TList [CBOR.TInt x, CBOR.TBool diffusionMode, CBOR.TInt peerSharing])
+          | x >= 0
+          , x <= 0xffffffff
+          , peerSharing >= 0
+          , peerSharing <= 2
+          = Right
+              NodeToNodeVersionData {
+                  networkMagic = NetworkMagic (fromIntegral x),
+                  diffusionMode = if diffusionMode
+                                  then InitiatorOnlyDiffusionMode
+                                  else InitiatorAndResponderDiffusionMode,
+                  peerSharing = case peerSharing of
+                                  0 -> NoPeerSharing
+                                  1 -> PeerSharingPrivate
+                                  2 -> PeerSharingPublic
+                                  _ -> error "decodeTerm: impossible happened!",
+                  query = False
+                }
+          | x < 0 || x > 0xffffffff
+          = Left $ T.pack $ "networkMagic out of bound: " <> show x
+          | otherwise -- peerSharing < 0 || peerSharing > 2
+          = Left $ T.pack $ "peerSharing out of bound: " <> show peerSharing
+        decodeTerm _ t
+          = Left $ T.pack $ "unknown encoding: " ++ show t
+     in CodecCBORTerm {encodeTerm, decodeTerm = decodeTerm version }
```

Comment:
Is duplication necessary? can't one just do `query = query && version >= 12`?

Reply:
That would work for `decode` but not for `encode`; this way will be also easier when we will remove unused versions after a HF.

Reply:
... for `encode` one would also need an `if` to not include the `TInt` in the list.